### ISSUE TITLE
[SOLVED ] freeze after plymouth-start.services/dracut cleanup

### ISSUE THREAD

**Issue by SPIDR5WEB** [ublue-os/bazzite#227]
Hey guys. Would love to run your OS but I get this is the error when I try to install either bazzite or ublue. I tried all obvious and apparent solutions from using multiple usbs, switching between recommended image writers, as well as search the web but I found little documentation. Maybe a piece of hardware isn't supported or I don't have my bios set up properly?  I am familiar with other Linux flavors but this is my first time installing a fedora os so if anyone knows anything, I'd appreciate the help! 

Desktop i5, Nvidia gtx 1660ti, msi Mobo, 32gb ram.

![20230829_230108](https://github.com/ublue-os/bazzite/assets/143575707/19d1818f-dc19-4f98-a953-6a34684d8561)

**Comment by rodnheli** [comment]
[https://github.com/ublue-os/bazzite/issues/109#issuecomment-1694484745](url)

**Comment by SPIDR5WEB** [comment]
> 

I used Rufus and multiple pc's to write it to multiple thumb drives. Even the universal blue os got stuck here. I even checked the USB drives for corruption.

**Comment by SPIDR5WEB** [comment]
update: I was even able to install Fedora workstation on my pc, rewrite using fedora media writer on fedora, and still get the same screen.

**Comment by kubepy** [comment]
What is the logs when append the kernel argument with `rd.debug` ?

**Comment by kubepy** [comment]
`initrd.img` has the service `/etc/systemd/system/initrd.target.wants/dracut-pre-pivot.service`

I just guessed maybe the following one of service has the errors, that `dracut-pre-pivot.service` cannot continue.
```
After=initrd.target initrd-parse-etc.service sysroot.mount
After=dracut-initqueue.service dracut-pre-mount.service dracut-mount.service
```

```
cat etc/systemd/system/initrd.target.wants/dracut-pre-pivot.service 
#  This file is part of dracut.
#
# See dracut.bootup(7) for details

[Unit]
Description=dracut pre-pivot and cleanup hook
Documentation=man:dracut-pre-pivot.service(8)
DefaultDependencies=no
After=initrd.target initrd-parse-etc.service sysroot.mount
After=dracut-initqueue.service dracut-pre-mount.service dracut-mount.service
Before=initrd-cleanup.service
Wants=remote-fs.target
After=remote-fs.target
ConditionPathExists=/usr/lib/initrd-release
ConditionDirectoryNotEmpty=|/lib/dracut/hooks/pre-pivot
ConditionDirectoryNotEmpty=|/lib/dracut/hooks/cleanup
ConditionKernelCommandLine=|rd.break=pre-pivot
ConditionKernelCommandLine=|rd.break=cleanup
ConditionKernelCommandLine=|rd.break
ConditionPathExists=|/dev/root
ConditionPathExists=|/dev/nfs
Conflicts=shutdown.target emergency.target

[Service]
Environment=DRACUT_SYSTEMD=1
Environment=NEWROOT=/sysroot
Type=oneshot
ExecStart=-/bin/dracut-pre-pivot
StandardInput=null
StandardError=journal+console
KillMode=process
RemainAfterExit=yes
KeyringMode=shared

# Bash ignores SIGTERM, so we send SIGHUP instead, to ensure that bash
# terminates cleanly.
KillSignal=SIGHUP
```

**Comment by kubepy** [comment]
Hope the `rd.debug` argument could help you find the root cause.

**Comment by kubepy** [comment]
See [https://man7.org/linux/man-pages/man7/dracut.bootup.7.html](https://man7.org/linux/man-pages/man7/dracut.bootup.7.html)

If `dracut-pre-pivot.service` is finished, go to `initrd-cleanup.service`

```
$ cat etc/systemd/system/initrd.target.wants/dracut-pre-pivot.service | grep Before
Before=initrd-cleanup.service
```

```
$ cat usr/lib/systemd/system/initrd-cleanup.service 
#  SPDX-License-Identifier: LGPL-2.1-or-later
#
#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.

[Unit]
Description=Cleaning Up and Shutting Down Daemons
DefaultDependencies=no
AssertPathExists=/etc/initrd-release
OnFailure=emergency.target
OnFailureJobMode=replace-irreversibly
After=initrd-root-fs.target initrd-fs.target initrd.target

[Service]
Type=oneshot
ExecStart=systemctl --no-block isolate initrd-switch-root.target
```

and 
```
After=initrd-root-fs.target initrd-fs.target initrd.target
```

```
ls -1 etc/systemd/system/initrd.target.wants/*
etc/systemd/system/initrd.target.wants/dracut-cmdline-ask.service
etc/systemd/system/initrd.target.wants/dracut-cmdline.service
etc/systemd/system/initrd.target.wants/dracut-initqueue.service
etc/systemd/system/initrd.target.wants/dracut-mount.service
etc/systemd/system/initrd.target.wants/dracut-pre-mount.service
etc/systemd/system/initrd.target.wants/dracut-pre-pivot.service
etc/systemd/system/initrd.target.wants/dracut-pre-trigger.service
etc/systemd/system/initrd.target.wants/dracut-pre-udev.service
etc/systemd/system/initrd.target.wants/nm-initrd.service
etc/systemd/system/initrd.target.wants/nm-wait-online-initrd.service
```

**Comment by kubepy** [comment]
If you try to append the `rd.debug rd.break=cleanup rd.shell` parameters may see the command line shell, 
I don't know whether use `systemctl list-jobs` or `systemctl --failed` command could find the answer.

**Comment by SPIDR5WEB** [comment]
> If you try to append the `rd.debug rd.break=cleanup rd.shell` parameters may see the command line shell, I don't know whether use `systemctl list-jobs` or `systemctl --failed` command could find the answer.

sorry, I attempted some of your suggestions and got it to load into the installer but it freezes at 'deployment starting: ghcr.io/ublue-os/bazzite-nvidia:38'

**Comment by kubepy** [comment]
> sorry, I attempted some of your suggestions and got it to load into the installer but it freezes at 'deployment starting: ghcr.io/ublue-os/bazzite-nvidia:38'

is that networking issue when pulling the docker image same as #189 ?

**Comment by SPIDR5WEB** [comment]
> > sorry, I attempted some of your suggestions and got it to load into the installer but it freezes at 'deployment starting: ghcr.io/ublue-os/bazzite-nvidia:38'
> 
> is that networking issue when pulling the docker image same as #189 ?

It was actually just hanging on that screen while installing not any kind of network problem, my bad. Currently I was able to use your suggestion of adding `rd.debug rd.shell` to the kernal arguments right before rd.livecheck which pushed the process through and successfully booted into the fedora installer where I was able to install Bazzite! This worked on both of the Nvidia versions. II was mistaken when I thought it was frozen at `deployment starting: ghcr.io/ublue-os/bazzite-nvidia:38` the loading bar just stays still for a long period of time until it's finished installing. 

You were a great help and I think it may be a solid solution to:

https://github.com/ublue-os/bazzite/issues/109

In summary; If you highlight an Nvidia version of bazzite in grub and press E, you will load into a screen where you can add and delete arguments. If you add `rd.debug rd.shell` before or after `rd.livecheck` you will see debugging information and it should NOT freeze on dracut! You should load into the installer and proceed to install Bazzite-OS!

Thank You Kubepy!

**Comment by kubepy** [comment]
@SPIDR5WEB 
NP, sounds good and glad to see you have solved the issue.

**Comment by juicypop** [comment]
I'm also stuck on installing either nvidia version.

I wanted to try adding these commands, but there is no `rd.livecheck` when i hit `e` to edit the grub commands.
I tried the online 1.3 and offline 1.1 isos, both have no  `rd.livecheck` in their grub commands.

**Comment by noelmiller** [comment]
We are currently in the process of regenerating this ISO. It is not complete yet. If you want more information, please follow this issue: https://github.com/ublue-os/main/issues/468
